src/client/src/ui/analytics/views/positions-chart/positionsBubbleChart.js

```diff
@@ -0,0 +1,219 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import d3 from 'd3';
+import d3tip from 'd3-tip';
+import _ from 'lodash';
+import Dimensions from 'react-dimensions';
+import { logger } from '../../../../system';
+import {  createScales,
+          getPositionsDataFromSeries,
+          updateNodes,
+          drawCircles,
+          drawLabels,
+          getRadius,
+          getPositionValue} from './chartUtil';
+
+const _log:logger.Logger = logger.create('PositionsBubbleChart');
+
+@Dimensions()
+export default class PositionsBubbleChart extends React.Component{
+
+  static propTypes = {
+    data: React.PropTypes.array,
+    containerWidth: React.PropTypes.number,
+    containerHeight: React.PropTypes.number
+  };
+
+  static defaultProps = {
+    data: []
+  };
+
+  constructor(props){
+    super(props);
+    this.state = {
+      nodes: [],
+      prevPositionsData: {}
+    };
+  }
+
+  componentDidMount() {
+    this._redrawChart();
+  };
+
+  componentWillReceiveProps(nextProps){
+    if (this._shouldRedrawChart(nextProps)){
+      this._redrawChart(nextProps);
+    }
+    this._isUpdateRequired(nextProps);
+  }
+
+  componentDidUpdate(){
+    this._update(this.state.nodes);
+  }
+
+  _shouldRedrawChart(nextProps = this.props){
+
+    if (this.state.nodes.length === 0 && nextProps.data.length > 0){
+      this._updateNodes(nextProps.data);
+    }
+
+    let positionsData = getPositionsDataFromSeries(nextProps.data);
+    let existingPositionsData = this.state.prevPositionsData;
+    let nodesChanged = positionsData.length !== existingPositionsData.length;
+
+    if (nodesChanged){
+      this.setState({updateRequired: true});
+    }
+    return nodesChanged;
+  }
+
+  _isUpdateRequired(nextProps = this.props){
+    let positionsData = getPositionsDataFromSeries(nextProps.data);
+    let existingPositionsData = this.state.prevPositionsData;
+
+    //positions data has changed on the existing nodes
+    let modifiedData = _.reduce(positionsData, function(result, value, key) {
+      return _.isEqual(value, existingPositionsData[key]) ?
+        result : result.concat(key);
+    }, []);
+
+    let stalePositions = _.filter(existingPositionsData, (existingPos) => _.findIndex(positionsData, (pos) => pos.symbol === existingPos.symbol) === -1);
```

Comment:
would rather do a _.map and then filter, instead of doing a findIndex for each pos